sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -134,6 +131,32 @@ def fit(self, X, y=None):
 
         return self
 
+    def add_element(self, f, v, feature_names, vocab, fitting=True,
+                    transforming=False, indices=None, values=None):
+        if isinstance(v, six.string_types):
+            feature_name = "%s%s%s" % (f, self.separator, v)
+            v = 1
+        elif isinstance(v, Number) or (v is None):
+            feature_name = f
+        elif isinstance(v, Iterable):
+            for vv in v:
+                self.add_element(f, vv, feature_names, vocab,
```

Comment:
This will flatten a nested list. Is there a reason we would want that? Is there a rain we would not want that?

Reply:
The nested list case should, at least, be tested, for either support or failure